src/core/bpf-firewall.c

```diff
@@ -539,6 +543,52 @@ int bpf_firewall_compile(Unit *u) {
         if (r < 0)
                 return log_unit_error_errno(u, r, "Compilation for egress BPF program failed: %m");
 
+        if (supported == BPF_FIREWALL_SUPPORTED_WITH_MULTI) { /* Skip silently otherwise */
+                STRV_FOREACH(bpf_fs_path, cc->ip_filters_ingress) {
+                        r = bpf_program_item_new(&prog_item);
+                        if (r < 0)
+                                return log_unit_error_errno(u, r, "Can't allocate BPF program list entry: %m");
+
+                        r = bpf_program_new(BPF_PROG_TYPE_CGROUP_SKB, &prog_item->prog);
+                        if (r < 0)
+                                return log_unit_error_errno(u, r, "Can't allocate CGROUP SKB BPF program: %m");
+
+                        r = bpf_program_load_from_bpf_fs(prog_item->prog, *bpf_fs_path);
+                        if (r < 0)
+                                return log_unit_error_errno(u, r, "Loading of ingress BPF program %s failed: %m", *bpf_fs_path);
+
+                        if (!u->ip_bpf_custom_ingress) {
+                                r = bpf_program_list_new(&u->ip_bpf_custom_ingress);
+                                if (r < 0)
+                                        return log_unit_error_errno(u, r, "Can't allocate BPF program list: %m");
+                        }
+                        LIST_APPEND(items, u->ip_bpf_custom_ingress->list, prog_item);
+                        prog_item = NULL;
+                }
+
+                STRV_FOREACH(bpf_fs_path, cc->ip_filters_egress) {
+                        r = bpf_program_item_new(&prog_item);
+                        if (r < 0)
+                                return log_unit_error_errno(u, r, "Can't allocate BPF program list entry: %m");
+
+                        r = bpf_program_new(BPF_PROG_TYPE_CGROUP_SKB, &prog_item->prog);
+                        if (r < 0)
+                                return log_unit_error_errno(u, r, "Can't allocate CGROUP SKB BPF program: %m");
+
+                        r = bpf_program_load_from_bpf_fs(prog_item->prog, *bpf_fs_path);
+                        if (r < 0)
+                                return log_unit_error_errno(u, r, "Loading of egress BPF program %s failed: %m", *bpf_fs_path);
+
+                        if (!u->ip_bpf_custom_egress) {
+                                r = bpf_program_list_new(&u->ip_bpf_custom_egress);
+                                if (r < 0)
+                                        return log_unit_error_errno(u, r, "Can't allocate BPF program list: %m");
+                        }
+                        LIST_APPEND(items, u->ip_bpf_custom_egress->list, prog_item);
+                        prog_item = NULL;
+                }
```

Comment:
this looks like like these two loops could be unified in one, if they split out into a function that is called twice, once with ip_filters_ingress and once with ip_filters_egress...